scripts/generate_package_diffs_for_flavors/generate_package_diffs_for_flavors.py

```diff
@@ -0,0 +1,290 @@
+            diff_df = compare_package_lists(package_list_working_copy_str_2, package_list_working_copy_str_1)
+            diff_df = diff_df.rename(columns={"Version1": f"Version in {working_copy_2_name}",
+                                              "Version2": f"Version in {working_copy_1_name}"})
+            result[result_key] = diff_df
+    return result
+
+
+def compare_flavor(flavor_path_1: Path, working_copy_1: Path, working_copy_1_name: str,
+                   flavor_path_2: Path, working_copy_2: Path, working_copy_2_name: str) \
+        -> Dict[Tuple[str, str], Dict[Tuple[str, Optional[str]], pd.DataFrame]]:
+    flavor_base_path_1 = Path(flavor_path_1, "flavor_base")
+    flavor_base_path_2 = Path(flavor_path_2, "flavor_base")
+    result = {}
+    if flavor_base_path_1.is_dir():
+        for build_step_path_1 in flavor_base_path_1.iterdir():
+            if build_step_path_1.is_dir():
+                build_step_name_1 = build_step_path_1.name
+                build_step_name_2 = build_step_name_1
+                build_step_path_2 = Path(flavor_base_path_2, build_step_name_2)
+                diffs = compare_build_step(build_step_path_1, working_copy_1, working_copy_1_name,
+                                           build_step_path_2, working_copy_2, working_copy_2_name)
+                result[(build_step_name_1, build_step_name_2)] = diffs
+    return result
+
+
+def get_last_git_tag() -> str:
+    get_last_tag_command = ["git", "describe", "--abbrev=0", "--tags"]
+    last_tag_result = subprocess.run(get_last_tag_command, stdout=subprocess.PIPE)
+    last_tag_result.check_returncode()
+    last_tag = last_tag_result.stdout.decode("UTF-8").strip()
+    return last_tag
+
+
+def checkout_git_tag_as_worktree(tmp_dir, last_tag):
+    checkout_last_tag_command = ["git", "worktree", "add", tmp_dir, last_tag]
+    checkout_last_tag_result = subprocess.run(checkout_last_tag_command, stderr=subprocess.PIPE)
+    checkout_last_tag_result.check_returncode()
+    init_submodule_command = ["git", "submodule", "update", "--init"]
+    init_submodule_result = subprocess.run(init_submodule_command, cwd=tmp_dir, stderr=subprocess.PIPE)
+    init_submodule_result.check_returncode()
+
+
+def generate_dependency_diff_report_for_package_list(
+        package_file_diff_file: Path,
+        diff_df: pd.DataFrame):
+    package_file_diff_file.parent.mkdir(parents=True, exist_ok=True)
+    with package_file_diff_file.open("wt") as f:
+        diff_df.to_markdown(f)
+
+
+def generate_dependency_diff_report_for_build_step(
+        build_steps: Tuple[str, str],
+        diffs: Dict[Tuple[str, Optional[str]], pd.DataFrame],
+        base_output_directory: Path,
+        relative_output_directory: Path):
+    result = ""
+    if len(diffs) > 0:
+        build_step_caption = generate_build_step_caption(build_steps)
+        result = f"- {build_step_caption}\n"
```

Comment:
Maybe it would be better to use some library to assemble the markdown content.

Reply:
For example https://github.com/didix21/mdutils#create-markdown-files

Reply:
sounds good, but looking at the time, I would create a ticket for it and do it after the release.

Reply:
https://github.com/exasol/script-languages-release/issues/321